network/src/commonMain/kotlin/com/wire/kalium/network/NetworkClient.kt

```diff
@@ -104,37 +145,32 @@ internal sealed class HttpClientOptions {
  * @param options, some configuration presets
  * @param config, a block that allows further customisation of the [HttpClient]
  */
-internal fun provideBaseHttpClient(
-    engine: HttpClientEngine, options: HttpClientOptions, config: HttpClientConfig<*>.() -> Unit = {}
-) = HttpClient(engine) {
-    install(DefaultRequest)
+private fun HttpClientConfig<*>.installWireBaseUrl(backend: ServerConfigDTO) {
     defaultRequest {
-
-        // since error response are application/json
-        // this header is added by default to all requests
         header(HttpHeaders.ContentType, ContentType.Application.Json)
-
-        when (options) {
-            HttpClientOptions.NoDefaultHost -> {/* do nothing */
-            }
-
-            is HttpClientOptions.DefaultHost -> {
-                with(options.serverConfigDTO) {
-                    // enforce https as url protocol
-                    url.protocol = URLProtocol.HTTPS
-                    // add the default host
-                    url.host = apiBaseUrl.host
-
-                    // for api version 0 no api version should be added to the request
-                    url.encodedPath =
-                        if (shouldAddApiVersion(apiVersion)) apiBaseUrl.encodedPath + "v${apiVersion}/"
-                        else apiBaseUrl.encodedPath
-                }
-            }
+        with(backend) {
+            val apiBaseUrl = backend.links.api
+            // enforce https as url protocol
+            url.protocol = URLProtocol.HTTPS
+            // add the default host
+            url.host = apiBaseUrl.host
+            // for api version 0 no api version should be added to the request
+            url.encodedPath =
+                if (shouldAddApiVersion(metaData.commonApiVersion.version))
+                    apiBaseUrl.encodedPath + "v${metaData.commonApiVersion.version}/"
+                else apiBaseUrl.encodedPath
         }
     }
+}
 
-    if (NetworkLogger.isRequestLoggingEnabled) {
+internal fun provideBaseHttpClient(
+    engine: HttpClientEngine, config: HttpClientConfig<*>.() -> Unit = {}
+) = HttpClient(engine) {
+    install(UserAgent) {
+        agent = "007"
```

Comment:
Where's the 🔫?

For real tho, should we remove this? What was @tmspzz's recommendation for `User-Agent` header?